### PR TITLE
Add an option to load only from a branch

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue  #210.

#### Summary of Changes
add a new option to filter PR's only from a branch

#### Additional comments
maybe should be better to properly use the Github API

### CODE REVIEW

File: administrator/components/com_patchtester/PatchTester/Model/PullsModel.php [hunk 1]
```diff
@@ -370,7 +371,9 @@ public function requestFromGithub($page)
 			return array('complete' => true);
 		}
 
-		$data = array();
+		$data     = array();
+		$params   = ComponentHelper::getParams('com_patchtester');
```
Comment (mbabker) [inline administrator/components/com_patchtester/PatchTester/Model/PullsModel.php]: This should be pushed into the model from the controller via the state data.

**Comment by mbabker** [comment]
OK, looking at this again, what is really being gained here?  We don't limit the number of API calls to GitHub and there is already a display filter for branches in the current code.

**Comment by alikon** [comment]
yes, it should be better fetched directly from the github API

**Comment by franz-wohlkoenig** [comment]
@mbabker to download what is needed.

Test 3.8, 3.9, 4.0 on different Installation and download same PR on each Installation.

**Comment by franz-wohlkoenig** [comment]
so you can't avoid downloading unnecessary PR?

**Comment by mbabker** [comment]
Right now, no, there is no way to avoid downloading extra data.  There is a way to change the API query, but as I said it has the side effect of removing the ability to fetch data without saving your GitHub credentials to the component configuration.  I don't want to remove that feature right now.

**Comment by alikon** [comment]
as stated, without changing the github lookup no, with this pr we can only avoid to store "unnecessary" pr's on db

**Comment by franz-wohlkoenig** [comment]
got it, thanks Guys for Info. So its easiest to stay as is.

**Pull request closed without merging**